AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -158,10 +158,16 @@ final class AWSInitialSyncOrchestrator: InitialSyncOrchestrator {
             return .successfulVoid
         }
 
+        var underlyingError: Error?
+        if syncErrors.contains(where: isNetworkError) {
+            underlyingError = getFirstUnderlyingNetworkError(errors: syncErrors)
```

Comment:
Any potential side effects on using the first network error? (ie. can different network errors have different retry policies?)

Reply:
nit: can we use `.first(where predicate:)` and check if `syncErrors` has any network error and grab the first in a single operation?

Reply:
Currently, it is not ideal to have RemoteSyncEngine contain URLError based retry logic, and this is why it's forcing us to propagate one of the network errors up to the RemoteSyncEngine. Ideally, InitialSyncOperation should be responsible for checking the network error, determining that the error is retryable, and retry with the exponential back-off. This means it will cycle on the single InitialSyncOperation trying to sync that one model. Once the network is back up, then the operation will finish, and move onto the next InitialSyncOperation.

This code current assumes that the first network error is probably similar to other network errors if they were run approximately the same time. Of course this assumption could be wrong, but the side effect of it being wrong and pick up a network error that RemoteSyncEngine decides is non-retryable is that the sync engine will move to an errored state, as with all other non-retryable errors.

The retry policy i believe is determined by the implementation in `RequestRetryablePolicy` which filters down to
```
case .notConnectedToInternet,
             .dnsLookupFailed,
             .cannotConnectToHost,
             .cannotFindHost,
             .timedOut:
```